lib/rule-tester/rule-tester.js

```diff
@@ -1258,7 +1279,12 @@ class RuleTester {
                         this.constructor[valid.only ? "itOnly" : "it"](
                             sanitize(typeof valid === "object" ? valid.name || valid.code : valid),
                             () => {
-                                testValidTemplate(valid);
+                                try {
+                                    runHook(valid, "before");
+                                    testValidTemplate(valid);
+                                } finally {
+                                    runHook(valid, "after");
+                                }
```

Comment:
ℹ️ this approach allows to run the `after()` hook anyway.
errors thrown by `before()` and by the test case itself are not being catched (no `catch` block), so that the workflow operation remains intact.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch